### ISSUE TITLE
Security Keys don't work/will be detected from Brave

### ISSUE THREAD

**Issue by EinfachNurBaum** [brave/brave-browser#33237]
## Description 
If Websites like Google asks for security key, Brvae can not detect it. It works fine on Firefox. The Nitrokey and Yubikey does not work. I tried USB-A and USB-C port

## Expected result:
It should detect my security keys.

## Brave version (brave://version info)
1.58.131 Chromium: 117.0.5938.92 (Offizieller Build) unknown (64-Bit)

## Miscellaneous Information:

Betriebssystem (OS)| Linux
JavaScript | V8 11.7.439.16
User-Agent | Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/117.0.0.0 Safari/537.36
Befehlszeile | /app/brave/brave --disable-features=WebAssemblyTrapHandler,DesktopPWAsRunOnOsLogin --no-default- 
browser-check --disable-domain-reliability --enable-dom-distiller --origin-trial-public-key=bYUKPJoPnCxeNvu72j4EmPuK7tr1PAC7SHh8ld9Mw3E=,fMS4mpO6buLQ/QMd+zJmxzty/VQ6B1EUZqoCU04zoRU= --sync-url=https://sync-v2.brave.com/v2 --lso-url=https://no-thanks.invalid --variations-server-url=https://variations.brave.com/seed --variations-insecure-server-url=https://variations.brave.com/seed --disable-features=DesktopPWAsRunOnOsLogin,WebAssemblyTrapHandler --flag-switches-begin --flag-switches-end --component-updater=url-source=https://go-updater

**Comment by rebron** [comment]
cc: @fmarier

**Comment by GoLD-ReaVeR** [comment]
Confirming I got the same issue here trying to log in to azure. The key's strings seem to be completely ignored and instead the enter button pushed by the key is being sent to the UI triggering the "Use different method" option bringing me back where I started.

**Comment by samuel-lucas6** [comment]
If you go to _Try WebAuthn_ on https://demo.yubico.com/, do you get the following?

> The operation either timed out or was not allowed. See: https://www.w3.org/TR/webauthn-2/#sctn-privacy-considerations-client.

Just received my YubiKey, and it's not working in Brave with this error but works in Microsoft Edge. #34441 is related. I've tried adjusting my settings and creating a new profile, but neither fixed the problem.

**Comment by samuel-lucas6** [comment]
I've tested my YubiKey on macOS with the latest/same version of Brave and it works, but I still can't get it to work on Windows.

**Comment by joelgriffiths** [comment]
This thread is one of the top results on Google. I'm having the same symptoms. Works with Chrome, but not Brave.

**Comment by fmarier** [comment]
Can anybody share the URLs of the websites where your YubiKeys don't work? That way we could try to reproduce on Windows. We're not aware of general problems with security keys, so we'd need to investigate further.

Also, it would be useful to know which YubiKey model you have.

**Comment by joelgriffiths** [comment]
The Yukico.com verification website.

https://www.yubico.com/genuine/

**Comment by joelgriffiths** [comment]
Also, I was able to confirm it works with Linux as well. Just not Windows.

**Comment by samuel-lucas6** [comment]
@fmarier

> Can anybody share the URLs of the websites where your YubiKeys don't work? That way we could try to reproduce on Windows. We're not aware of general problems with security keys, so we'd need to investigate further.

https://demo.yubico.com/ (WebAuthn)
https://www.yubico.com/genuine/

The only account I tested trying to set up was GitHub (Settings, Password and authentication, Security keys), and it gave a `Security key registration failed.` error.

> Also, it would be useful to know which YubiKey model you have.

The YubiKey 5 NFC. Sorry for not mentioning that.

**Comment by joelgriffiths** [comment]
Doesn't work with this one either. Same error. Chrome and Firefox are fine. I've disabled all extensions and rebooted. No love.

https://www.amazon.com/dp/B0BVNPWPCN?psc=1&ref=ppx_yo2ov_dt_b_product_details

**Comment by localh0st** [comment]
Attempting to register the YubiKey 5 NFC at the demo site: [https://demo.yubico.com/webauthn-technical/registration](https://demo.yubico.com/webauthn-technical/registration)

A browser dialog pops up with "Create a Passkey".  Selecting "Windows Hello or External Security Key" results in the following error:

> The operation either timed out or was not allowed. See: https://www.w3.org/TR/webauthn-2/#sctn-privacy-considerations-client.

In other browsers, including Chrome, the same selection leads to a Windows dialog that allows registration of the key.  Perhaps Windows is blocking Brave from accessing Windows Security Center?

Running Brave v1.64.109, Windows 11 Pro 22H2 Build 22621.2428

**Comment by samuel-lucas6** [comment]
> A browser dialog pops up with "Create a Passkey". Selecting "Windows Hello or External Security Key" results in the following error

Interestingly, I don't even get a dialog pop up.

**Comment by fmarier** [comment]
It looks like this may affect only version 1.64 on Windows. Can anybody confirm whether or not this works for them on Brave Beta (1.65)?

**Comment by MadhaviSeelam** [comment]
Able to reproduce the issue in `1.64.109 Chromium: 123.0.6312.58` with the site https://www.yubico.com/genuine/.  But works just fine `1.65.93 Chromium: 123.0.6312.58` 

1.64.109 | 1.65.93
---- | -----
<img width="909" alt="image" src="https://github.com/brave/brave-browser/assets/98358127/9a83e49f-5807-4106-bcec-735264b896ea">|<img width="956" alt="image" src="https://github.com/brave/brave-browser/assets/98358127/282e7d1b-2f9e-4865-a5e5-5dc45a753bc1">

**Comment by localh0st** [comment]
Confirmed, @fmarier. 

[https://demo.yubico.com/webauthn-technical/registration](https://demo.yubico.com/webauthn-technical/registration) does work properly with Brave (Beta) Version 1.65.89 Chromium: 123.0.6312.58

**Comment by joelgriffiths** [comment]
Yes, it does work properly for the registration website. However, it doesn't read at all when trying to login to the LastPass extension. Touch the key, and it just sits there.

**Comment by samuel-lucas6** [comment]
@fmarier Can also confirm that both links I listed work with Brave Beta 1.65.

**Comment by EthanHeilman** [comment]
@samuel-lucas6 I can confirm that it is broken on 1.64.109 and that it is fixed in 1.66.39 

See my compatibility matrix here:
https://github.com/brave/brave-browser/issues/37023

**Comment by EthanHeilman** [comment]
Given that this appears to be fixed in the nightly build, do we know why it broke? What change caused this bug?

**Comment by fmarier** [comment]
> Given that this appears to be fixed in the nightly build, do we know why it broke? What change caused this bug?

Not yet, we're still trying to figure it out because the changes we've looked at so far (between a version known to be broken and one known to work) haven't turned up anything obvious.

**Comment by joesixpack** [comment]
> Confirmed, [@fmarier](https://github.com/fmarier).
> 
> https://demo.yubico.com/webauthn-technical/registration does work properly with Brave (Beta) Version 1.65.89 Chromium: 123.0.6312.58

Part 2 fails on Brave v1.82.166 @ Windows 11 Build 26100.6584.

<img width="454" height="340" alt="Image" src="https://github.com/user-attachments/assets/8d0d4d0e-ab82-4315-a2ea-072e67298e12" />

**Comment by kabakaev** [comment]
How many of you have a password manager extension, which hijacks the passkey/WebAuthn flows and then fails silently?

At least it was my case today with Bitwarden extension, which has `Bitwarden` -> `Settings` -> `Notifications` -> `Ask to save and use passkeys` option enabled by default, causing the passkey WebAuthn failure.
